src/server/game/Instances/InstanceScript.cpp

```diff
@@ -415,10 +415,19 @@ bool InstanceScript::SetBossState(uint32 id, EncounterState state)
                     break;
                 }
                 case FAIL:
+                {
                     ResetCombatResurrections();
                     SendEncounterEnd();
+
+                    instance->DoOnPlayers([](Player* player)
+                    {
+                        if (player->IsAlive())
+                            player->RemoveAurasWithInterruptFlags(SpellAuraInterruptFlags2::EndOfEncounter);
```

Comment:
Not sure if we should check IsAlive() for this, might as well leave it outside of the check since we don't even do that in Battleground::RemovePlayerAtLeave to remove battleground-specific auras.